aii-pxelinux/src/main/perl/pxelinux.pm

```diff
@@ -233,257 +303,338 @@ sub pxe_network_bonding {
             }
         }
 
         if($kst->{bootproto} eq 'static') {
             if ($ksdev =~ m/^((?:(?:[0-9a-f]{2}[:-]){5}[0-9a-f]{2})|bootif|link)$/i) {
-                $this_app->error("Invalid ksdevice $ksdev for static ks configuration.");
+                $self->error("Invalid ksdevice $ksdev for static ks configuration.");
             } else {
-                my $static = pxe_ks_static_network($cfg, $ksdev);
-                push(@append,"ip=$static") if ($static);
+                my $static = $self->_pxe_ks_static_network($cfg, $ksdev);
+                push(@kernel_params,"ip=$static") if ($static);
             }
         } elsif ($kst->{bootproto} =~ m/^(dhcp6?|auto6|ibft)$/) {
-            push(@append,"ip=$kst->{bootproto}");
+            push(@kernel_params,"ip=$kst->{bootproto}");
         }
 
         my $nms = $cfg->getElement("/system/network/nameserver")->getTree;
         foreach my $ns (@$nms) {
-            push(@append,"nameserver=$ns");
+            push(@kernel_params,"nameserver=$ns");
         }
     }
 
-    my $custom_append = $t->{append};
+    my $custom_append = $pxe_config->{append};
     if ($custom_append) {
 	    $custom_append =~ s/LOCALHOST/$server/g;
-	    push @append, $custom_append;
+	    push @kernel_params, $custom_append;
     }
     
-    return @append;    
+    return @kernel_params;    
 }
 
-# create a list with all append options
-sub pxe_append
+# create a list with all required kernel parameters, based on the configuration
+sub _kernel_params
 {
-    my $cfg = shift;
+    my ($self, $cfg, $variant) = @_;
 
     if ($cfg->elementExists(KS)) {
-        return pxe_ks_append($cfg);
+        return $self->_kernel_params_ks($cfg, $variant);
     } else {
-        $this_app->error("Unclear how to create the append options. Not using any options.");
+        $self->error("No Kickstart-related parameters in configuration: no kernel parameters added.");
         return;
     }
 }
 
-# Prints the PXE configuration file.
-sub pxeprint
+# Write the PXELINUX configuration file.
+sub _write_pxelinux_config
 {
-    my $cfg = shift;
-    my $t = $cfg->getElement (PXEROOT)->getTree;
-    my $fh = CAF::FileWriter->open (filepath ($cfg),
-				    log => $this_app, mode => 0644);
+    my ($self, $cfg) = @_;
+    my $pxe_config = $cfg->getElement (PXEROOT)->getTree;
+    my $fh = CAF::FileWriter->open ($self->_file_path ($cfg, PXE_VARIANT_PXELINUX),
+                    log => $self, mode => 0644);
 
     my $appendtxt = '';
-    my @appendoptions = pxe_append($cfg);
+    my @appendoptions = $self->_kernel_params($cfg, PXE_VARIANT_PXELINUX);
     $appendtxt = join(" ", "append", @appendoptions) if @appendoptions;
 
-    $fh->print (<<EOF
+    my $entry_label = "Install $pxe_config->{label}";
+    print $fh <<EOF;
 # File generated by pxelinux AII plug-in.
 # Do not edit.
-default $t->{label}
-    label $t->{label}
-    kernel $t->{kernel}
+default $entry_label
+    label $entry_label
```

Comment:
I'm happy to do it! I have not done it before because I wanted to minimize the changes to the PXELINUX config file and this format is the one generated for quite some time. But I agree that it is misleading... and led me to the error fixed in the commit (with the label not maching what was on the default line.